pkg/plugin/settings.go

```diff
@@ -186,6 +187,19 @@ func LoadSettings(ctx context.Context, config backend.DataSourceInstanceSettings
 		}
 	}
 
+	settings.EnableRowLimit = false
+	if jsonData["enableRowLimit"] != nil {
+		if enableRowLimitString, ok := jsonData["enableRowLimit"].(string); ok {
+			settings.EnableRowLimit, err = strconv.ParseBool(enableRowLimitString)
+			if err != nil {
+				backend.Logger.Warn("Failed to parse enableRowLimit value, defaulting to false", "error", err)
+				settings.EnableRowLimit = false
+			}
+		} else {
+			settings.EnableRowLimit = jsonData["enableRowLimit"].(bool)
```

Comment:
are we sure of the type of the `enableRowLimit` key? if it's an int, this will [panic](https://go.dev/play/p/G8xWI-4cvqE).

Reply:
righto.